File: ContestOfSwift/TestsOfSwift/Challenges/Challenge5Tests.swift
//
//  Challenge5Tests.swift
//  TestsOfSwift
//
//  Created by Sergio Daniel on 4/26/24.
//

import Foundation
import XCTest

final class Challenge5Tests: XCTestCase {
    
    func testTriangle() {
        let triangle1 = Challenge5.Triangle(base: 20, height: 20)
        XCTAssertEqual(
            triangle1.getArea(),
            200
        )
        XCTAssertTrue(triangle1.isEquilateral)
        XCTAssertFalse(triangle1.isRectangular)
        XCTAssertTrue(triangle1.isIsosceles)

        let triangle2 = Challenge5.Triangle(base: 30, height: 30, firstAngle: 90)
        XCTAssertEqual(
            triangle2.getArea(),
            450
        )
        XCTAssertTrue(triangle2.isRectangular)
        XCTAssertFalse(triangle2.isEquilateral)
        XCTAssertTrue(triangle2.isIsosceles)

        let triangle3 = Challenge5.Triangle(base: 30, height: 20, firstAngle: 30)
        XCTAssertEqual(
            triangle3.getArea(),
            300
        )
        XCTAssertFalse(triangle3.isRectangular)
        XCTAssertFalse(triangle3.isEquilateral)
        XCTAssertFalse(triangle3.isIsosceles)
    }

    func testRectangle() {
        let rectangle1 = Challenge5.Rectangle(length: 30, width: 20)
        XCTAssertEqual(
            rectangle1.getArea(),
            600
        )
        XCTAssertFalse(rectangle1.isEquilateral)
        XCTAssertFalse(rectangle1.isSquare)

        let rectangle2 = Challenge5.Rectangle(length: 30)
        XCTAssertEqual(
            rectangle2.getArea(),
            900
        )
        XCTAssertTrue(rectangle2.isEquilateral)
        XCTAssertTrue(rectangle2.isSquare)
    }

}

File: ContestOfSharp/TestOfSharp/Challenges/Challenge5Tests.cs
using ContestOfSharp.Challenges;

namespace TestOfSharp.Challenges;

[TestClass]
public class Challenge5Tests
{
    [TestMethod]
    public void TestTriangles()
    {
        var triangle1 = new Challenge5.Triangle() { Base = 20, Height = 20 };
        Assert.AreEqual(200, triangle1.GetArea());
        Assert.IsTrue(triangle1.IsEquilateral);
        Assert.IsFalse(triangle1.IsRectangular);
        Assert.IsTrue(triangle1.IsIsosceles);

        var triangle2 = new Challenge5.Triangle(firstAngle: 90.0) { Base = 30, Height = 30 };
        Assert.AreEqual(450, triangle2.GetArea());
        Assert.IsFalse(triangle2.IsEquilateral);
        Assert.IsTrue(triangle2.IsRectangular);
        Assert.IsFalse(triangle2.IsIsosceles);

        var triangle3 = new Challenge5.Triangle(firstAngle: 30) { Base = 30, Height = 20 };
        Assert.AreEqual(300, triangle3.GetArea());
        Assert.IsFalse(triangle3.IsEquilateral);
        Assert.IsFalse(triangle3.IsRectangular);
        Assert.IsFalse(triangle3.IsIsosceles);
    }

    [TestMethod]
    public void TestRectangles()
    {
        var rectangle1 = new Challenge5.Rectangle(length: 30, width: 20);
        Assert.AreEqual(600, rectangle1.GetArea());
        Assert.IsFalse(rectangle1.IsEquilateral);
        Assert.IsFalse(rectangle1.IsSquare);

        var rectangle2 = new Challenge5.Rectangle(length: 30, width: 30);
        Assert.AreEqual(900, rectangle2.GetArea());
        Assert.IsTrue(rectangle2.IsEquilateral);
        Assert.IsTrue(rectangle2.IsSquare);
    }
}